### PR TITLE
Add gpu pool to test deployment and enable gpu in sample test

### PR DESCRIPTION
**Pull request by hongye-sun**
I temporarily disable the clean up code for debug purpose. Will update the PR once the test passed.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/696)

### CODE REVIEW

**Comment by hongye-sun** [comment]
@gaoning777, the PR is ready, PTAL.

File: test/sample-test/run_test.sh [hunk 1]
```diff
@@ -152,6 +152,7 @@ if [ "$TEST_NAME" == 'tf-training' ]; then
     sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_PREDICT_IMAGE}|g" kubeflow-training-classification.py
     sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" kubeflow-training-classification.py
   fi
+  sed -i -e "s|use_gpu\s*=\s*False|use_gpu = True|g" kubeflow-training-classification.py
```
Comment (vicaire) [inline test/sample-test/run_test.sh]: That seems a bit hacky. Could we use proper parameter passing instead of using sed on a Python file?

Comment (hongye-sun) [reply test/sample-test/run_test.sh]: We did sed replace all over the places in the test script. There is currently a limitation that user cannot pass gpu setting as pipeline parameter as gpu must be set before the dsl compile time. That's why we cannot use parameter to control this test.

Comment (vicaire) [reply test/sample-test/run_test.sh]: OK, thanks.

**Pull request merged**